redis/src/main/scala/zio/redis/api/SortedSets.scala

```diff
@@ -368,40 +473,71 @@ trait SortedSets {
+      Tuple4(ArbitraryInput[K](), ArbitraryInput[String](), ArbitraryInput[String](), OptionalInput(LimitInput)),
+      ChunkOutput(ArbitraryOutput[M]())
     )
-    command.run((key, lexRange, limit))
+    command.run((key, lexRange.max.stringify, lexRange.min.stringify, limit))
   }
 
   /**
    * Return a range of members in a sorted set, by score, with scores ordered from high to low.
    *
    * @param key Key of a sorted set
    * @param scoreRange ScoreRange that must be satisfied
-   * @param withScores The optional WITHSCORES argument makes the command return both the element and its score, instead of the element alone
    * @param limit The optional LIMIT argument can be used to only get a range of the matching elements. A negative count returns all elements from the offset
-   * @return Chunk of elements in the specified range (optionally with their scores)
+   * @return Chunk of elements in the specified range
    */
-  final def zRevRangeByScore(
-    key: String,
+  final def zRevRangeByScore[K: Schema, M: Schema](
+    key: K,
     scoreRange: ScoreRange,
-    withScores: Option[WithScores] = None,
     limit: Option[Limit] = None
-  ): ZIO[RedisExecutor, RedisError, Chunk[String]] = {
+  ): ZIO[RedisExecutor, RedisError, Chunk[M]] = {
     val command = RedisCommand(
       ZRevRangeByScore,
-      Tuple4(StringInput, ScoreRangeInput, OptionalInput(WithScoresInput), OptionalInput(LimitInput)),
-      ChunkOutput(MultiStringOutput)
+      Tuple4(
+        ArbitraryInput[K](),
+        ArbitraryInput[String](),
+        ArbitraryInput[String](),
+        OptionalInput(LimitInput)
+      ),
+      ChunkOutput(ArbitraryOutput[M]())
+    )
+    command.run((key, scoreRange.max.stringify, scoreRange.min.stringify, limit))
+  }
+
+  /**
+   * Return a range of members in a sorted set, by score, with scores ordered from high to low.
+   *
+   * @param key Key of a sorted set
+   * @param scoreRange ScoreRange that must be satisfied
+   * @param limit The optional LIMIT argument can be used to only get a range of the matching elements. A negative count returns all elements from the offset
+   * @return Chunk of elements with their scores in the specified range
+   */
+  final def zRevRangeByScoreWithScores[K: Schema, M: Schema](
+    key: K,
+    scoreRange: ScoreRange,
+    limit: Option[Limit] = None
+  ): ZIO[RedisExecutor, RedisError, Chunk[MemberScore[M]]] = {
+    val command = RedisCommand(
+      ZRevRangeByScore,
+      Tuple5(
+        ArbitraryInput[K](),
+        ArbitraryInput[String](),
+        ArbitraryInput[String](),
+        ArbitraryInput[String](),
+        OptionalInput(LimitInput)
+      ),
+      ChunkTuple2Output(ArbitraryOutput[M](), ArbitraryOutput[Double]())
+        .map(_.map { case (m, s) => MemberScore(s, m) })
     )
-    command.run((key, scoreRange, withScores, limit))
+    command.run((key, scoreRange.max.stringify, scoreRange.min.stringify, WithScores.stringify, limit))
```

Comment:
I wonder if it would be better to also have a `ScoreRangeInput` like we have for example for `MemberScore`

Reply:
As you can see, I've deleted ScoreRangeInput, LexRangeInput because if not than we need RevScoreRangeInput and RevLexRangeInput that would have different position of max and min.
In my opinion it better construct result output from ADT output that we have already, then write new ones. But may be I'm wrong, may be it would be nice to discuss that with community?
@dmijic What is your opinion about that?)  
P.S. Actually, It fine for me do it in both ways.

Reply:
Yea also true. I think we are going to iterate Input and Output types anyways. So I would say it if fine for now as it is

Reply:
I agree with @anatolysergeev . In my opinion, I only define input if necessary, because they simply wrap parameters. For some options, I will use it.